system_tests/bigquery.py

```diff
@@ -288,6 +288,58 @@ def _has_rows(result):
         self.assertEqual(sorted(rows, key=by_age),
                          sorted(ROWS, key=by_age))
 
+    def test_load_table_from_local_file_then_dump_table(self):
+        import csv
+        import tempfile
+        local_id = unique_resource_id()
+        BUCKET_NAME = 'bq_load_test' + local_id
+        BLOB_NAME = 'person_ages.csv'
+        GS_URL = 'gs://%s/%s' % (BUCKET_NAME, BLOB_NAME)
+        ROWS = [
+            ('Phred Phlyntstone', 32),
+            ('Bharney Rhubble', 33),
+            ('Wylma Phlyntstone', 29),
+            ('Bhettye Rhubble', 27),
+        ]
+        TABLE_NAME = 'test_table'
+
+        dataset = Config.CLIENT.dataset(DATASET_NAME)
+
+        retry_403(dataset.create)()
+        self.to_delete.append(dataset)
+
+        full_name = bigquery.SchemaField('full_name', 'STRING',
+                                         mode='REQUIRED')
+        age = bigquery.SchemaField('age', 'INTEGER', mode='REQUIRED')
+        table = dataset.table(TABLE_NAME, schema=[full_name, age])
+        table.create()
+        self.to_delete.insert(0, table)
+
+        with tempfile.NamedTemporaryFile(mode='w+') as csv_file:
+            writer = csv.writer(csv_file)
+            writer.writerow(('Full Name', 'Age'))
+            writer.writerows(ROWS)
+            csv_file.flush()
+
+            with open(csv_file.name, 'rb') as csv_read:
```

Comment:
You could the `delete=False` option but then you would have to add delete logic at the end of the test or in a tearDown.